docs/source/mods/line-optimization.rst

```diff
@@ -0,0 +1,232 @@
+            1	1	3	5
+            2	1	4	25
+            3	1	5	10
+          >>> frequencies = [1,3]
+
+      For the example we used data of the Sioux-Falls network. It is not considered as a realistic one. However,
+      this network can be found on different websites when considering traffic problems
+      (originally by Hillel Bar-Gera http://www.bgu.ac.il/~bargera/tntp/). We added a set of line routes.
+      Note that the output shown above contains some additional information that is not required for computation, for example
+      the property length in the edge data.
+      Also, ``posx`` and ``posy`` in the ``node_data`` is not used for computation. But it can be used to visualize the
+      network as done below.
+      It is important that all data is consistant. For example, ``edgeSource``, ``edgeTarget``
+      in the ``linepath_data`` must correspond to a ``number`` in the node_data. The same holds
+      for ``source`` and ``target`` in ``edge_data`` and ``demand_data``.
+      In the code it is checked that all tables provide the relevant columns.
+      Note that the edges are assumed to be directed and both direction need to be defined if an edge
+      can be traversed in both directions. In the same way, a line is a directed path. If a line is
+      turning at the end point and goes back the same way, the nodes need to be added again in reverse order.
+
+Solution
+--------
+
+The solution consists of two information
+
+- the total cost of the optimal line plan
+- the optimal line plan as a list of linename-frequency tuples.
+
+The strategy can be defined via a Boolean parameter **shortestPaths**. This parameter has a default value (True)
+which uses approach 1, i.e., routing passengers on shortest paths only.
+Note that strategy 1 needs the python package networkx. If this is not available, the second approach is used.
+The second approach is also used if the parameter shortestPaths is set to False.
+
+.. tabs::
+
+  .. group-tab:: pandas
+
+      .. doctest:: pandas
+          :options: +NORMALIZE_WHITESPACE
+
+          >>> from gurobi_optimods import datasets
+          >>> from gurobi_optimods.line_optimization import line_optimization
+          >>> node_data, edge_data, line_data, linepath_data, demand_data = datasets.load_siouxfalls_network_data()
+          >>> frequencies = [1,3]
+          >>> objCost, finalLines = line_optimization(node_data, edge_data, line_data, linepath_data, demand_data, frequencies, True, verbose=False)
+          >>> objCost
+          211.0
+          >>> finalLines
+          [('new271_B', 1), ('new31_B', 1), ('new407_B', 1), ('new415_B', 3), ('new423_B', 3), ('new535_B', 3), ('new551_B', 3), ('new71_B', 1)]
+
+The Sioux-Falls transportation network (left) and the optimal line plan (right) for this example is shown in the figure below. The lines are shown as
+different colored paths in the network.
+
+.. image:: figures/lop_siouxfalls_solution.png
+  :width: 600
+  :alt: SiouxFalls.
```

Comment:
Do you have code to produce this plot from the solution? If it's relatively simple, it would be good to include it as part of the example. If it's complex, we could think about including a plotting function like we have in the OPF optimod (if you think it's worthwhile).

Reply:
I added a plot function. However, if lines share the same link, it is not clear how to plot them in a nice way. For Siouxfalls it was good enough to add an offset to the x- and y-coordinate. I am not sure how this looks for other networks.

Reply:
Could you please check if the plot function is worth to be added?

Reply:
This is great! Yes I think this is definitely worth including, having the visualization is a nice addition.